diff -ur bind-9.2.4-orig/configure.in bind-9.2.4/configure.in
--- bind-9.2.4-orig/configure.in	2004-09-24 12:08:40.000000000 +0200
+++ bind-9.2.4/configure.in	2004-09-24 12:09:25.417209496 +0200
@@ -899,8 +899,8 @@
 	mips-sgi-irix*)
 		;;
 	*)
-		AC_CHECK_LIB(socket, socket)
-		AC_CHECK_LIB(nsl, inet_ntoa)
+		AC_SEARCH_LIBS(socket, socket)
+		AC_SEARCH_LIBS(inet_ntoa, nsl)
 		;;
 esac
 
diff -ur bind-9.2.4-orig/lib/bind/configure.in bind-9.2.4/lib/bind/configure.in
--- bind-9.2.4-orig/lib/bind/configure.in	2004-09-24 12:08:40.000000000 +0200
+++ bind-9.2.4/lib/bind/configure.in	2004-09-24 12:09:25.429207672 +0200
@@ -664,9 +664,9 @@
 	mips-sgi-irix*)
 		;;
 	*)
-		AC_CHECK_LIB(d4r, gethostbyname_r)
-		AC_CHECK_LIB(socket, socket)
-		AC_CHECK_LIB(nsl, inet_ntoa)
+		AC_SEARCH_LIBS(gethostbyname_r, d4r)
+		AC_SEARCH_LIBS(socket, socket)
+		AC_SEARCH_LIBS(inet_ntoa, nsl)
 		;;
 esac
 
diff -ur bind-9.2.4-orig/lib/bind/Makefile.in bind-9.2.4/lib/bind/Makefile.in
--- bind-9.2.4-orig/lib/bind/Makefile.in	2004-07-20 09:00:18.000000000 +0200
+++ bind-9.2.4/lib/bind/Makefile.in	2004-09-24 12:09:25.415209800 +0200
@@ -19,6 +19,8 @@
 VPATH =         @srcdir@
 top_srcdir =    @top_srcdir@
 
+LIBS = @LIBS@
+
 @LIBBIND_API@
 
 LIBS =		@LIBS@
diff -ur bind-9.2.4-orig/lib/dns/Makefile.in bind-9.2.4/lib/dns/Makefile.in
--- bind-9.2.4-orig/lib/dns/Makefile.in	2004-07-20 09:00:19.000000000 +0200
+++ bind-9.2.4/lib/dns/Makefile.in	2004-09-24 12:10:43.106398944 +0200
@@ -38,7 +38,7 @@
 
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la -lcrypto
 
 # Alphabetically
 
@@ -169,7 +169,7 @@
 	./gen -s ${srcdir} > code.h
 
 gen: gen.c
-	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
+	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o $@ ${srcdir}/gen.c
 
 rbtdb64.@O@: rbtdb.c
 
diff -ur bind-9.2.4-orig/lib/isccc/Makefile.in bind-9.2.4/lib/isccc/Makefile.in
--- bind-9.2.4-orig/lib/isccc/Makefile.in	2004-07-20 09:00:20.000000000 +0200
+++ bind-9.2.4/lib/isccc/Makefile.in	2004-09-24 12:09:25.422208736 +0200
@@ -36,7 +36,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCCDEPLIBS =	libisccc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la
 
 SUBDIRS =	include
 
diff -ur bind-9.2.4-orig/lib/isccfg/Makefile.in bind-9.2.4/lib/isccfg/Makefile.in
--- bind-9.2.4-orig/lib/isccfg/Makefile.in	2004-07-20 09:00:20.000000000 +0200
+++ bind-9.2.4/lib/isccfg/Makefile.in	2004-09-24 12:09:25.424208432 +0200
@@ -38,7 +38,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCFGDEPLIBS =	libisccfg.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la ../dns/libdns.la
 
 SUBDIRS =	include
 
diff -ur bind-9.2.4-orig/lib/lwres/Makefile.in bind-9.2.4/lib/lwres/Makefile.in
--- bind-9.2.4-orig/lib/lwres/Makefile.in	2004-08-28 08:15:27.000000000 +0200
+++ bind-9.2.4/lib/lwres/Makefile.in	2004-09-24 12:09:25.426208128 +0200
@@ -44,7 +44,8 @@
 		lwres_gabn.c lwres_gnba.c lwres_grbn.c lwres_noop.c \
 		lwinetaton.c lwinetpton.c lwinetntop.c print.c
 
-LIBS =		@LIBS@
+# -lpthread not needed here
+LIBS =
 
 SUBDIRS =	include man unix
 TARGETS =	timestamp
