--- bind-9.2.1/lib/bind/Makefile.in~	Thu Aug  9 05:59:34 2001
+++ bind-9.2.1/lib/bind/Makefile.in	Fri Nov 29 16:55:09 2002
@@ -19,6 +19,8 @@
 VPATH =         @srcdir@
 top_srcdir =    @top_srcdir@
 
+LIBS = @LIBS@
+
 @LIBBIND_API@
 
 DAEMON_OBJS=bsd/daemon.@O@
--- bind-9.2.1/configure.in.orig	Wed Dec 25 00:03:03 2002
+++ bind-9.2.1/configure.in	Wed Dec 25 00:33:51 2002
@@ -768,8 +768,8 @@
 	mips-sgi-irix*)
 		;;
 	*)
-		AC_CHECK_LIB(socket, socket)
-		AC_CHECK_LIB(nsl, inet_ntoa)
+		AC_SEARCH_LIBS(socket, socket)
+		AC_SEARCH_LIBS(inet_ntoa, nsl)
 		;;
 esac
 
--- bind-9.2.1/lib/dns/Makefile.in.orig	Thu Nov 15 02:24:06 2001
+++ bind-9.2.1/lib/dns/Makefile.in	Wed Dec 25 00:37:47 2002
@@ -30,7 +30,7 @@
 CDEFINES =
 CWARNINGS =
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la -lcrypto
 
 # Alphabetically
 
@@ -140,7 +140,7 @@
 	./gen -s ${srcdir} > code.h
 
 gen: gen.c
-	${CC} ${ALL_CFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
+	${CC} ${ALL_CFLAGS} -o $@ ${srcdir}/gen.c
 
 rbtdb64.@O@: rbtdb.c
 
--- bind-9.2.1/lib/isccc/Makefile.in.orig	Tue Jul 10 00:25:26 2001
+++ bind-9.2.1/lib/isccc/Makefile.in	Wed Dec 25 00:39:15 2002
@@ -36,7 +36,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCCDEPLIBS =	libisccc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la
 
 SUBDIRS =	include
 
--- bind-9.2.1/lib/isccfg/Makefile.in.orig	Tue Jul 10 00:25:27 2001
+++ bind-9.2.1/lib/isccfg/Makefile.in	Wed Dec 25 00:41:21 2002
@@ -36,7 +36,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCFGDEPLIBS =	libisccfg.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la ../dns/libdns.la
 
 SUBDIRS =	include
 
--- bind-9.2.1/lib/lwres/Makefile.in.orig	Tue Jul 10 20:25:37 2001
+++ bind-9.2.1/lib/lwres/Makefile.in	Wed Dec 25 00:30:35 2002
@@ -44,7 +44,8 @@
 		lwres_gabn.c lwres_gnba.c lwres_grbn.c lwres_noop.c \
 		lwinetaton.c lwinetpton.c lwinetntop.c
 
-LIBS =		@LIBS@
+# -lpthread not needed here
+LIBS =
 
 SUBDIRS =	include man unix
 TARGETS =	timestamp
--- bind-9.2.1/lib/bind/configure.in.orig	Wed Dec 25 01:35:07 2002
+++ bind-9.2.1/lib/bind/configure.in	Wed Dec 25 02:03:15 2002
@@ -662,9 +662,9 @@
 	mips-sgi-irix*)
 		;;
 	*)
-		AC_CHECK_LIB(d4r, gethostbyname_r)
-		AC_CHECK_LIB(socket, socket)
-		AC_CHECK_LIB(nsl, inet_ntoa)
+		AC_SEARCH_LIBS(gethostbyname_r, d4r)
+		AC_SEARCH_LIBS(socket, socket)
+		AC_SEARCH_LIBS(inet_ntoa, nsl)
 		;;
 esac
 
