diff -Naur bind-9.1.0/lib/isc/unix/socket.c bind-9.1.0-p/lib/isc/unix/socket.c
--- bind-9.1.0/lib/isc/unix/socket.c	Thu Jan 18 00:42:41 2001
+++ bind-9.1.0-p/lib/isc/unix/socket.c	Mon Mar 26 03:37:21 2001
@@ -2733,6 +2733,7 @@
 		backlog = SOMAXCONN;
 
 	if (listen(sock->fd, (int)backlog) < 0) {
+		close(sock->fd);
 		UNLOCK(&sock->lock);
 		UNEXPECTED_ERROR(__FILE__, __LINE__, "listen: %s",
 				 strerror(errno));
