diff -ur src.old/lib/bsd/Makefile src/lib/bsd/Makefile
--- src.old/lib/bsd/Makefile	Mon Feb 22 03:47:57 1999
+++ src/lib/bsd/Makefile	Mon May 24 15:24:40 1999
@@ -79,7 +79,7 @@
 	-rmdir ${THREADED}
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] .
diff -ur src.old/lib/cylink/Makefile src/lib/cylink/Makefile
--- src.old/lib/cylink/Makefile	Mon Feb 22 03:47:57 1999
+++ src/lib/cylink/Makefile	Mon May 24 15:24:58 1999
@@ -81,7 +81,7 @@
 	-rmdir ${THREADED}
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] .
diff -ur src.old/lib/dnssafe/Makefile src/lib/dnssafe/Makefile
--- src.old/lib/dnssafe/Makefile	Tue Feb 23 06:26:59 1999
+++ src/lib/dnssafe/Makefile	Mon May 24 15:24:13 1999
@@ -98,7 +98,7 @@
 	-rmdir ${THREADED}
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] .
diff -ur src.old/lib/dst/Makefile src/lib/dst/Makefile
--- src.old/lib/dst/Makefile	Sun Mar  7 10:33:47 1999
+++ src/lib/dst/Makefile	Mon May 24 15:25:19 1999
@@ -83,7 +83,7 @@
 	-rmdir ${THREADED}
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] SRC/*.pl .
diff -ur src.old/lib/inet/Makefile src/lib/inet/Makefile
--- src.old/lib/inet/Makefile	Wed Mar  3 09:07:16 1999
+++ src/lib/inet/Makefile	Mon May 24 15:27:19 1999
@@ -80,7 +80,7 @@
 
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] .
diff -ur src.old/lib/irs/Makefile src/lib/irs/Makefile
--- src.old/lib/irs/Makefile	Mon Feb 22 03:47:58 1999
+++ src/lib/irs/Makefile	Mon May 24 15:27:48 1999
@@ -100,7 +100,7 @@
 	-rmdir ${THREADED}
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] .
diff -ur src.old/lib/isc/Makefile src/lib/isc/Makefile
--- src.old/lib/isc/Makefile	Mon Feb 22 03:47:58 1999
+++ src/lib/isc/Makefile	Mon May 24 15:28:15 1999
@@ -82,7 +82,7 @@
 	-rmdir ${THREADED}
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] .
diff -ur src.old/lib/nameser/Makefile src/lib/nameser/Makefile
--- src.old/lib/nameser/Makefile	Mon Feb 22 03:47:59 1999
+++ src/lib/nameser/Makefile	Mon May 24 15:28:32 1999
@@ -75,7 +75,7 @@
 	-rmdir ${THREADED}
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] .
diff -ur src.old/lib/resolv/Makefile src/lib/resolv/Makefile
--- src.old/lib/resolv/Makefile	Mon Feb 22 03:47:59 1999
+++ src/lib/resolv/Makefile	Mon May 24 15:28:50 1999
@@ -79,7 +79,7 @@
 	-rmdir ${THREADED}
 
 depend: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	../../port/linux/bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] .
diff -ur src.old/port/linux/Makefile src/port/linux/Makefile
--- src.old/port/linux/Makefile	Mon Feb 22 03:48:02 1999
+++ src/port/linux/Makefile	Mon May 24 15:22:43 1999
@@ -67,7 +67,7 @@
 	rm -f *.${O} *.BAK *.CKP *~
 
 depend:: FRC
-	mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
+	bin/mkdep -I${INCL} -I${PORTINCL} ${CPPFLAGS} ${SRCS}
 
 links: FRC
 	@set -e; ln -s SRC/*.[ch] SRC/bin SRC/Makefile.set SRC/probe .
