diff -urN bind-9.0.1rc2.org/configure.in bind-9.0.1rc2/configure.in
--- bind-9.0.1rc2.org/configure.in	Tue Nov  7 11:04:02 2000
+++ bind-9.0.1rc2/configure.in	Wed Nov  8 11:15:29 2000
@@ -184,12 +184,13 @@
 #
 AC_MSG_CHECKING(for compatible OpenSSL library)
 
-DST_PRIVATEOPENSSL='-DDST_USE_PRIVATE_OPENSSL'
-dst_privateopenssl='openssl'
-DST_OPENSSL_INC='-I${srcdir}/../openssl/include'
+DNS_OPENSSL_LIBS="-lssl -lcrypto"
+DST_PRIVATEOPENSSL=''
+dst_privateopenssl=''
+DST_OPENSSL_INC=''
 DST_OPENSSL_LIB=''
-DST_OPENSSL_OBJS='${OPENSSLOBJS}'
-AC_MSG_RESULT(using private library)
+DST_OPENSSL_OBJS=''
+AC_MSG_RESULT(using shared library)
 
 AC_SUBST(DST_PRIVATEOPENSSL)
 AC_SUBST(dst_privateopenssl)
@@ -202,7 +203,6 @@
 # it as needed) if it is found.
 #
 
-DNS_OPENSSL_LIBS=""
 AC_SUBST(DNS_OPENSSL_LIBS)
 
 #
