--- bind-9.3.1/contrib/idn/idnkit-1.0-src/patch/bind9/bind-9.3.1-patch~	2004-12-13 01:06:44.000000000 +0100
+++ bind-9.3.1/contrib/idn/idnkit-1.0-src/patch/bind9/bind-9.3.1-patch	2005-03-15 11:53:31.726411768 +0100
@@ -995,7 +995,7 @@
 +				     sizeof(utf8_textname));
 +		idn_check_result(mr, "append origin to textname");
 +	}
-+	mr = idn_encodename(IDN_LOCALMAP | IDN_NAMEPREP | IDN_ASCCHECK |
++	mr = idn_encodename(IDN_LOCALMAP | IDN_NAMEPREP | 
 +			    IDN_IDNCONV | IDN_LENCHECK, utf8_textname,
 +			    idn_textname, sizeof(idn_textname));
 +	idn_check_result(mr, "convert UTF-8 textname to IDN encoding");
