diff -ur bind-9.2.5.orig/lib/bind/Makefile.in bind-9.2.5/lib/bind/Makefile.in
--- bind-9.2.5.orig/lib/bind/Makefile.in	2005-03-23 23:06:54.000000000 +0100
+++ bind-9.2.5/lib/bind/Makefile.in	2005-03-23 23:09:36.000000000 +0100
@@ -100,11 +100,11 @@
 
 @BIND9_MAKE_RULES@
 
-libbind.@SA@: ${OBJS}
+libbind.@SA@: touchfile
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libbind.la: ${OBJS}
+libbind.la: touchfile
 	${LIBTOOL_MODE_LINK} \
 	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libbind.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
diff -ur bind-9.2.5.orig/lib/bind/make/rules.in bind-9.2.5/lib/bind/make/rules.in
--- bind-9.2.5.orig/lib/bind/make/rules.in	2004-10-20 02:14:53.000000000 +0200
+++ bind-9.2.5/lib/bind/make/rules.in	2005-03-23 23:09:36.000000000 +0100
@@ -49,8 +49,10 @@
 ### Makefile may define:
 ###	TARGETS
 
-all: subdirs ${TARGETS}
+all: touchfile ${TARGETS}
 
+touchfile: subdirs
+	touch touchfile
 ###
 ### Subdirectories
 ###
diff -ur bind-9.2.5.orig/lib/dns/Makefile.in bind-9.2.5/lib/dns/Makefile.in
--- bind-9.2.5.orig/lib/dns/Makefile.in	2005-03-23 23:06:54.000000000 +0100
+++ bind-9.2.5/lib/dns/Makefile.in	2005-03-23 23:09:36.000000000 +0100
@@ -107,11 +107,11 @@
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libdns.@SA@: ${OBJS}
+libdns.@SA@: touchfile ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libdns.la: ${OBJS}
+libdns.la: touchfile ${OBJS}
 	${LIBTOOL_MODE_LINK} \
 		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libdns.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
@@ -137,7 +137,7 @@
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${GSSAPIOBJS} ${LIBS}
 
-timestamp: libdns.@A@
+timestamp: include libdns.@A@
 	touch timestamp
 
 installdirs:
diff -ur bind-9.2.5.orig/lib/isc/Makefile.in bind-9.2.5/lib/isc/Makefile.in
--- bind-9.2.5.orig/lib/isc/Makefile.in	2004-07-20 09:00:19.000000000 +0200
+++ bind-9.2.5/lib/isc/Makefile.in	2005-03-23 23:09:36.000000000 +0100
@@ -88,11 +88,11 @@
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libisc.@SA@: ${OBJS}
+libisc.@SA@: touchfile ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libisc.la: ${OBJS}
+libisc.la: touchfile ${OBJS}
 	${LIBTOOL_MODE_LINK} \
 		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisc.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
diff -ur bind-9.2.5.orig/make/rules.in bind-9.2.5/make/rules.in
--- bind-9.2.5.orig/make/rules.in	2004-07-20 09:00:21.000000000 +0200
+++ bind-9.2.5/make/rules.in	2005-03-23 23:09:36.000000000 +0100
@@ -47,7 +47,10 @@
 ### Makefile may define:
 ###	TARGETS
 
-all: subdirs ${TARGETS}
+all: touchfile ${TARGETS}
+
+touchfile: subdirs
+	touch touchfile
 
 ###
 ### Subdirectories
