diff -urN bind-9.2.1.org/lib/bind9/include/Makefile.in bind-9.2.1/lib/bind9/include/Makefile.in
--- bind-9.2.1.org/lib/bind9/include/Makefile.in	Sun Sep 29 00:00:18 2002
+++ bind-9.2.1/lib/bind9/include/Makefile.in	Sun Sep 29 00:01:05 2002
@@ -31,17 +31,17 @@
 @BIND9_MAKE_RULES@
 
 installdirs:
-	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${includedir} \
-	${DESTDIR}${includedir}/arpa ${DESTDIR}${includedir}/isc
+	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${includedir}/bind \
+	${DESTDIR}${includedir}/bind/arpa ${DESTDIR}${includedir}/bind/isc
 
 install:: installdirs
 	for i in ${HEADERS}; do \
-		${INSTALL_DATA} ${srcdir}/$$i ${DESTDIR}${includedir}; \
+		${INSTALL_DATA} ${srcdir}/$$i ${DESTDIR}${includedir}/bind; \
 	done
 	for i in ${IHEADERS}; do \
-		${INSTALL_DATA} ${srcdir}/$$i ${DESTDIR}${includedir}/isc; \
+		${INSTALL_DATA} ${srcdir}/$$i ${DESTDIR}${includedir}/bind/isc; \
 	done
 	for i in ${AHEADERS}; do \
-		${INSTALL_DATA} ${srcdir}/$$i ${DESTDIR}${includedir}/arpa; \
+		${INSTALL_DATA} ${srcdir}/$$i ${DESTDIR}${includedir}/bind/arpa; \
 	done
 

