diff -Nur bind-9.3.2.orig/configure.in bind-9.3.2.link/configure.in
--- bind-9.3.2.orig/configure.in	2006-08-05 01:45:48.000000000 +0200
+++ bind-9.3.2.link/configure.in	2006-08-05 01:45:55.000000000 +0200
@@ -1171,8 +1171,8 @@
 	*-linux*)
 		;;
 	*)
-		AC_CHECK_LIB(socket, socket)
-		AC_CHECK_LIB(nsl, inet_addr)
+		AC_SEARCH_LIBS(socket, socket)
+		AC_SEARCH_LIBS(inet_ntoa, nsl)
 		;;
 esac
 
diff -Nur bind-9.3.2.orig/lib/dns/Makefile.in bind-9.3.2.link/lib/dns/Makefile.in
--- bind-9.3.2.orig/lib/dns/Makefile.in	2004-12-09 05:07:15.000000000 +0100
+++ bind-9.3.2.link/lib/dns/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -39,7 +39,7 @@
 
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la -lcrypto
 
 # Alphabetically
 
diff -Nur bind-9.3.2.orig/lib/isccc/Makefile.in bind-9.3.2.link/lib/isccc/Makefile.in
--- bind-9.3.2.orig/lib/isccc/Makefile.in	2004-07-20 09:01:58.000000000 +0200
+++ bind-9.3.2.link/lib/isccc/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -36,7 +36,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCCDEPLIBS =	libisccc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la
 
 SUBDIRS =	include
 
diff -Nur bind-9.3.2.orig/lib/isccfg/Makefile.in bind-9.3.2.link/lib/isccfg/Makefile.in
--- bind-9.3.2.orig/lib/isccfg/Makefile.in	2004-07-20 09:01:58.000000000 +0200
+++ bind-9.3.2.link/lib/isccfg/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -38,7 +38,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCFGDEPLIBS =	libisccfg.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la ../dns/libdns.la
 
 SUBDIRS =	include
 
--- bind-9.10.0/lib/irs/Makefile.in~	2014-04-28 21:18:53.000000000 +0200
+++ bind-9.10.0/lib/irs/Makefile.in	2014-05-01 10:13:55.831613124 +0200
@@ -42,7 +42,7 @@
 		gai_strerror.c getaddrinfo.c getnameinfo.c \
 		resconf.c
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isccfg/libisccfg.la ../isc/libisc.la ../dns/libdns.la
 
 SUBDIRS =	include
 TARGETS =	timestamp
--- bind-9.12.0/lib/ns/Makefile.in~	2018-01-17 07:56:09.000000000 +0100
+++ bind-9.12.0/lib/ns/Makefile.in	2018-01-24 08:29:54.886677552 +0100
@@ -37,7 +37,7 @@ DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_
 
 DNSDEPLIBS =	../../lib/dns/libdns.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../dns/libdns.la
 
 # Alphabetically
 OBJS =		client.@O@ interfacemgr.@O@ lib.@O@ \
