Only in bind-9.2.4: bind-pmake.patch
diff -ur bind-9.2.4-orig/lib/bind/make/rules.in bind-9.2.4/lib/bind/make/rules.in
--- bind-9.2.4-orig/lib/bind/make/rules.in	2004-03-15 05:44:47.000000000 +0100
+++ bind-9.2.4/lib/bind/make/rules.in	2004-09-24 12:14:10.777828088 +0200
@@ -49,8 +49,10 @@
 ### Makefile may define:
 ###	TARGETS
 
-all: subdirs ${TARGETS}
+all: touchfile ${TARGETS}
 
+touchfile: subdirs
+	touch touchfile
 ###
 ### Subdirectories
 ###
diff -ur bind-9.2.4-orig/lib/bind/Makefile.in bind-9.2.4/lib/bind/Makefile.in
--- bind-9.2.4-orig/lib/bind/Makefile.in	2004-09-24 12:09:25.000000000 +0200
+++ bind-9.2.4/lib/bind/Makefile.in	2004-09-24 12:17:08.951741536 +0200
@@ -96,11 +96,11 @@
 
 @BIND9_MAKE_RULES@
 
-libbind.@SA@: ${OBJS}
+libbind.@SA@: touchfile
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libbind.la: ${OBJS}
+libbind.la: touchfile
 	${LIBTOOL_MODE_LINK} \
 	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libbind.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
diff -ur bind-9.2.4-orig/lib/dns/Makefile.in bind-9.2.4/lib/dns/Makefile.in
--- bind-9.2.4-orig/lib/dns/Makefile.in	2004-09-24 12:10:43.000000000 +0200
+++ bind-9.2.4/lib/dns/Makefile.in	2004-09-24 12:16:30.551579248 +0200
@@ -99,11 +99,11 @@
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libdns.@SA@: ${OBJS}
+libdns.@SA@: touchfile ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libdns.la: ${OBJS}
+libdns.la: touchfile ${OBJS}
 	${LIBTOOL_MODE_LINK} \
 		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libdns.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
@@ -129,7 +129,7 @@
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${GSSAPIOBJS} ${LIBS}
 
-timestamp: libdns.@A@
+timestamp: include libdns.@A@
 	touch timestamp
 
 installdirs:
@@ -177,5 +177,8 @@
 	include/dns/rdatastruct.h code.h
 subdirs: include/dns/enumtype.h include/dns/enumclass.h \
 	include/dns/rdatastruct.h code.h
+$(OBJS): include/dns/enumtype.h include/dns/enumclass.h \
+	include/dns/rdatastruct.h code.h
 ${DNSOBJS}: include/dns/enumtype.h include/dns/enumclass.h \
         include/dns/rdatastruct.h
+
diff -ur bind-9.2.4-orig/lib/isc/Makefile.in bind-9.2.4/lib/isc/Makefile.in
--- bind-9.2.4-orig/lib/isc/Makefile.in	2004-07-20 09:00:19.000000000 +0200
+++ bind-9.2.4/lib/isc/Makefile.in	2004-09-24 12:15:16.713804296 +0200
@@ -88,11 +88,11 @@
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libisc.@SA@: ${OBJS}
+libisc.@SA@: touchfile ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libisc.la: ${OBJS}
+libisc.la: touchfile ${OBJS}
 	${LIBTOOL_MODE_LINK} \
 		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisc.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
diff -ur bind-9.2.4-orig/make/rules.in bind-9.2.4/make/rules.in
--- bind-9.2.4-orig/make/rules.in	2004-07-20 09:00:21.000000000 +0200
+++ bind-9.2.4/make/rules.in	2004-09-24 12:14:10.788826416 +0200
@@ -47,7 +47,10 @@
 ### Makefile may define:
 ###	TARGETS
 
-all: subdirs ${TARGETS}
+all: touchfile ${TARGETS}
+
+touchfile: subdirs
+	touch touchfile
 
 ###
 ### Subdirectories
