--- bind-chroot-8.2.2_P5/src/bin/ndc/pathnames.h.orig	Mon Aug 14 05:05:16 2000
+++ bind-chroot-8.2.2_P5/src/bin/ndc/pathnames.h	Mon Aug 14 05:30:49 2000
@@ -21,54 +21,58 @@
 
 #include <paths.h>
 
+#ifndef _CHROOTDIR
+#define _CHROOTDIR	"/var/lib/named/chroot"
+#endif
+
 #ifndef _PATH_CONF
 #define _PATH_CONF	"/etc/named.conf"
 #endif
 
 #ifndef _PATH_DEBUG
-#define _PATH_DEBUG	"named.run"
+#define _PATH_DEBUG	_CHROOTDIR "/var/tmp/named.run"
 #endif
 
 #ifndef _PATH_DUMPFILE
-#define _PATH_DUMPFILE	"named_dump.db"
+#define _PATH_DUMPFILE	_CHROOTDIR "/var/tmp/named_dump.db"
 #endif
 
 #ifndef _PATH_NAMED
-#define _PATH_NAMED	"/usr/sbin/named"
+#define _PATH_NAMED	_CHROOTDIR "/usr/sbin/named"
 #endif
 
 #ifndef _PATH_PIDFILE
-#define _PATH_PIDFILE	"/var/run/named.pid"
+#define _PATH_PIDFILE	_CHROOTDIR "/var/run/named.pid"
 #endif
 
 #ifndef _PATH_NDCSOCK
-#define _PATH_NDCSOCK	"/var/run/ndc"
+#define _PATH_NDCSOCK	_CHROOTDIR "/var/run/ndc"
 #endif
 
 #ifndef _PATH_STATS
-#define _PATH_STATS	"named.stats"
+#define _PATH_STATS	_CHROOTDIR "/var/tmp/named.stats"
 #endif
 
 #ifndef _PATH_MEMSTATS
-#define _PATH_MEMSTATS	"named.memstats"
+#define _PATH_MEMSTATS	_CHROOTDIR "/var/tmp/named.memstats"
 #endif
 
 #ifndef _PATH_TMPXFER
-#define _PATH_TMPXFER	"xfer.ddt.XXXXXX"
+#define _PATH_TMPXFER	_CHROOTDIR "/var/tmp/xfer.ddt.XXXXXX"
 #endif
 
 #ifndef _PATH_XFER
-#define _PATH_XFER	"/usr/sbin/named-xfer"
+#define _PATH_XFER	_CHROOTDIR "/usr/sbin/named-xfer"
 #endif
 
 #ifndef _PATH_XFERTRACE
-#define _PATH_XFERTRACE	"xfer.trace"
+#define _PATH_XFERTRACE	_CHROOTDIR "/var/tmp/xfer.trace"
 #endif
 
 #ifndef _PATH_XFERDDT
-#define _PATH_XFERDDT	"xfer.ddt"
+#define _PATH_XFERDDT	_CHROOTDIR "/var/tmp/xfer.ddt"
 #endif
 
 #ifndef _PATH_DEVNULL
-#define _PATH_DEVNULL	"/dev/null"
+#define _PATH_DEVNULL	_CHROOTDIR "/dev/null"
 #endif
