--- bind-9.18.0/configure.ac~	2022-01-24 21:06:11.000000000 +0100
+++ bind-9.18.0/configure.ac	2022-01-27 13:36:04.136536700 +0100
@@ -474,7 +474,7 @@ AS_IF([test "$enable_geoip" = "yes"],
 	       [ # default
 		   AX_SAVE_FLAGS([maxminddb])
 		   MAXMINDDB_CFLAGS="-I$with_maxminddb/include"
-		   MAXMINDDB_LIBS="-L$with_maxminddb/lib"
+		   MAXMINDDB_LIBS=""
 		   CFLAGS="$CFLAGS $MAXMINDDB_CFLAGS"
 		   LIBS="$LIBS $MAXMINDDB_LIBS"
 		   AC_SEARCH_LIBS([MMDB_open], [maxminddb],
@@ -1295,7 +1295,7 @@ AS_CASE([$with_libidn2],
 	[no],	[],
 	[*],	[AX_SAVE_FLAGS([libidn2])
 		 LIBIDN2_CFLAGS="-I$with_libidn2/include"
-		 LIBIDN2_LIBS="-L$with_libidn2/lib"
+		 LIBIDN2_LIBS=""
 		 CFLAGS="$CFLAGS $LIBIDN2_CFLAGS"
 		 LIBS="$LIBS $LIBIDN2_LIBS"
 		 AC_CHECK_HEADERS([idn2.h],
