--- bind-9.8.0-P4/configure.in~	2011-08-03 11:46:24.326770259 +0200
+++ bind-9.8.0-P4/configure.in	2011-08-03 11:49:53.240770212 +0200
@@ -1076,7 +1076,7 @@ case "$use_gssapi" in
 		    then
 			    LIBS="$TRY_LIBS $ISC_OPENSSL_LIBS"
 		    else
-			    LIBS="-L$use_gssapi/lib $TRY_LIBS $ISC_OPENSSL_LIBS"
+			    LIBS="$TRY_LIBS $ISC_OPENSSL_LIBS"
 		    fi
 		    AC_MSG_CHECKING(linking as $TRY_LIBS)
 		    AC_TRY_LINK([
@@ -1420,7 +1420,7 @@
 				fi
 				;;
 			*)
-				DST_OPENSSL_LIBS="-L$use_openssl/lib -lcrypto"
+				DST_OPENSSL_LIBS="-lcrypto"
 				;;
 			esac
 		fi
@@ -1596,7 +1596,7 @@
 	*)
 		if test -f $kame_path/lib/libinet6.a; then
 			AC_MSG_RESULT($kame_path/lib/libinet6.a)
-			LIBS="-L$kame_path/lib -linet6 $LIBS"
+			LIBS="-linet6 $LIBS"
 		else
 			AC_MSG_ERROR([$kame_path/lib/libinet6.a not found.
 
@@ -2989,16 +2989,16 @@
 case "$use_libiconv" in
 yes)
 	if test X$prefix = XNONE ; then
-		iconvlib="-L/usr/local/lib -R/usr/local/lib -liconv"
+		iconvlib="-liconv"
 	else
-		iconvlib="-L$prefix/lib -R$prefix/lib -liconv"
+		iconvlib="-liconv"
 	fi
 	;;
 no)
 	iconvlib=
 	;;
 *)
-	iconvlib="-L$use_libiconv/lib -R$use_libiconv/lib -liconv"
+	iconvlib="-liconv"
 	;;
 esac

@@ -4801,7 +4801,7 @@ if test "no" != "$use_idnkit"; then
 	if test "no" != "$idnlib"; then
 		IDNKIT_LIBS="$idnlib $iconvlib"
 	else
-		IDNKIT_LIBS="-L$idnkit_path/lib -lidnkit $iconvlib"
+		IDNKIT_LIBS="-lidnkit $iconvlib"
 	fi
 fi
 AC_SUBST(IDNKIT_LIBS)
