--- bind-9.4.0/bin/dig/dighost.c~	2007-02-15 00:45:43.000000000 +0100
+++ bind-9.4.0/bin/dig/dighost.c	2007-05-11 14:39:54.098100849 +0200
@@ -1816,7 +1816,7 @@
 				     sizeof(utf8_textname));
 		idn_check_result(mr, "append origin to textname");
 	}
-	mr = idn_encodename(IDN_LOCALMAP | IDN_NAMEPREP | IDN_ASCCHECK |
+	mr = idn_encodename(IDN_LOCALMAP | IDN_NAMEPREP | /* IDN_ASCCHECK | */
 			    IDN_IDNCONV | IDN_LENCHECK, utf8_textname,
 			    idn_textname, sizeof(idn_textname));
 	idn_check_result(mr, "convert UTF-8 textname to IDN encoding");
