diff -Nur bind-9.3.2.orig/configure.in bind-9.3.2.link/configure.in
--- bind-9.3.2.orig/configure.in	2006-08-05 01:45:48.000000000 +0200
+++ bind-9.3.2.link/configure.in	2006-08-05 01:45:55.000000000 +0200
@@ -866,8 +866,8 @@
 	mips-sgi-irix*)
 		;;
 	*)
-		AC_CHECK_LIB(socket, socket)
-		AC_CHECK_LIB(nsl, inet_ntoa)
+		AC_SEARCH_LIBS(socket, socket)
+		AC_SEARCH_LIBS(inet_ntoa, nsl)
 		;;
 esac
 
diff -Nur bind-9.3.2.orig/lib/bind/configure.in bind-9.3.2.link/lib/bind/configure.in
--- bind-9.3.2.orig/lib/bind/configure.in	2006-08-05 01:45:48.000000000 +0200
+++ bind-9.3.2.link/lib/bind/configure.in	2006-08-05 01:45:55.000000000 +0200
@@ -599,9 +599,9 @@
 		AC_CHECK_LIB(nsl, inet_ntoa)
 		;;
 	*)
-		AC_CHECK_LIB(d4r, gethostbyname_r)
-		AC_CHECK_LIB(socket, socket)
-		AC_CHECK_LIB(nsl, inet_ntoa)
+		AC_SEARCH_LIBS(gethostbyname_r, d4r)
+		AC_SEARCH_LIBS(socket, socket)
+		AC_SEARCH_LIBS(inet_ntoa, nsl)
 		;;
 esac
 
diff -Nur bind-9.3.2.orig/lib/bind/make/rules.in bind-9.3.2.link/lib/bind/make/rules.in
--- bind-9.3.2.orig/lib/bind/make/rules.in	2004-10-20 02:14:47.000000000 +0200
+++ bind-9.3.2.link/lib/bind/make/rules.in	2006-08-05 01:49:13.000000000 +0200
@@ -95,6 +95,7 @@
 
 CC = 		@CC@
 CFLAGS =	@CFLAGS@
+LDFLAGS =	@LDFLAGS@
 STD_CINCLUDES =	@STD_CINCLUDES@
 STD_CDEFINES =	@STD_CDEFINES@
 STD_CWARNINGS =	@STD_CWARNINGS@
diff -Nur bind-9.3.2.orig/lib/bind/Makefile.in bind-9.3.2.link/lib/bind/Makefile.in
--- bind-9.3.2.orig/lib/bind/Makefile.in	2005-07-29 02:13:08.000000000 +0200
+++ bind-9.3.2.link/lib/bind/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -23,6 +23,8 @@
 .NOTPARALLEL:
 .NO_PARALLEL:
 
+LIBS = @LIBS@
+
 @LIBBIND_API@
 
 LIBS =		@LIBS@
diff -Nur bind-9.3.2.orig/lib/dns/Makefile.in bind-9.3.2.link/lib/dns/Makefile.in
--- bind-9.3.2.orig/lib/dns/Makefile.in	2004-12-09 05:07:15.000000000 +0100
+++ bind-9.3.2.link/lib/dns/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -39,7 +39,7 @@
 
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la -lcrypto
 
 # Alphabetically
 
diff -Nur bind-9.3.2.orig/lib/isccc/Makefile.in bind-9.3.2.link/lib/isccc/Makefile.in
--- bind-9.3.2.orig/lib/isccc/Makefile.in	2004-07-20 09:01:58.000000000 +0200
+++ bind-9.3.2.link/lib/isccc/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -36,7 +36,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCCDEPLIBS =	libisccc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la
 
 SUBDIRS =	include
 
diff -Nur bind-9.3.2.orig/lib/isccfg/Makefile.in bind-9.3.2.link/lib/isccfg/Makefile.in
--- bind-9.3.2.orig/lib/isccfg/Makefile.in	2004-07-20 09:01:58.000000000 +0200
+++ bind-9.3.2.link/lib/isccfg/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -38,7 +38,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCFGDEPLIBS =	libisccfg.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la ../dns/libdns.la
 
 SUBDIRS =	include
 
diff -Nur bind-9.3.2.orig/lib/lwres/Makefile.in bind-9.3.2.link/lib/lwres/Makefile.in
--- bind-9.3.2.orig/lib/lwres/Makefile.in	2005-06-10 01:54:32.000000000 +0200
+++ bind-9.3.2.link/lib/lwres/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -46,7 +46,8 @@
 		lwinetaton.c lwinetpton.c lwinetntop.c print.c \
 		strtoul.c
 
-LIBS =		@LIBS@
+# -lpthread not needed here
+LIBS =
 
 SUBDIRS =	include man unix
 TARGETS =	timestamp
