diff -Naur bind-9.2.0/configure.in bind-9.2.0-p/configure.in
--- bind-9.2.0/configure.in	Sun Nov 25 02:48:43 2001
+++ bind-9.2.0-p/configure.in	Tue Nov 27 12:38:26 2001
@@ -24,7 +24,7 @@
 AC_CANONICAL_HOST
 
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 AC_PROG_INSTALL
 
 AC_SUBST(STD_CINCLUDES)
--- bind-9.2.0-p/lib/bind/configure.in	Sun Nov 25 02:49:32 2001
+++ bind-9.2.0/lib/bind/configure.in	Tue Nov 27 12:42:13 2001
@@ -28,7 +28,7 @@
 AC_CANONICAL_HOST
 
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 AC_PROG_INSTALL
 
 AC_SUBST(STD_CINCLUDES)
