diff -urN src.org/lib/irs/irpmarshall.c src/lib/irs/irpmarshall.c
--- src.org/lib/irs/irpmarshall.c	Sun Dec 10 17:49:15 2000
+++ src/lib/irs/irpmarshall.c	Sun Dec 10 17:50:33 2000
@@ -90,7 +90,9 @@
 #include "port_after.h"
 
 
+#ifndef _GNU_SOURCE
 static char    *strndup(const char *str, size_t len);
+#endif
 static char   **splitarray(const char *buffer, const char *buffend, char delim);
 static int	joinarray(char * const * argv, char *buffer, char delim);
 static char    *getfield(char **res, size_t reslen, char **buffer, char delim);
@@ -1876,6 +1878,7 @@
  *
  */
 
+#ifndef _GNU_SOURCE
 static char *
 strndup(const char *str, size_t len) {
 	char *p = malloc(len + 1);
@@ -1886,6 +1889,7 @@
 	p[len] = 0x0;
 	return (p);
 }
+#endif
 
 #if WANT_MAIN
 
