diff -urN bind-9.2.3.org/lib/bind/make/rules.in bind-9.2.3/lib/bind/make/rules.in
--- bind-9.2.3.org/lib/bind/make/rules.in	2003-10-24 23:31:33.000000000 +0200
+++ bind-9.2.3/lib/bind/make/rules.in	2003-10-24 23:31:50.000000000 +0200
@@ -49,8 +49,10 @@
 ### Makefile may define:
 ###	TARGETS
 
-all: subdirs ${TARGETS}
+all: touchfile ${TARGETS}
 
+touchfile: subdirs
+	touch touchfile
 ###
 ### Subdirectories
 ###
diff -urN bind-9.2.3.org/lib/bind/Makefile.in bind-9.2.3/lib/bind/Makefile.in
--- bind-9.2.3.org/lib/bind/Makefile.in	2003-10-24 23:31:33.000000000 +0200
+++ bind-9.2.3/lib/bind/Makefile.in	2003-10-24 23:31:50.000000000 +0200
@@ -94,11 +94,11 @@
 
 @BIND9_MAKE_RULES@
 
-libbind.@SA@: ${OBJS}
+libbind.@SA@: touchfile
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libbind.la: ${OBJS}
+libbind.la: touchfile
 	${LIBTOOL} --mode=link \
 	${CC} ${ALL_CFLAGS} -o libbind.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
diff -urN bind-9.2.3.org/lib/dns/Makefile.in bind-9.2.3/lib/dns/Makefile.in
--- bind-9.2.3.org/lib/dns/Makefile.in	2003-10-24 23:31:34.000000000 +0200
+++ bind-9.2.3/lib/dns/Makefile.in	2003-10-24 23:32:39.000000000 +0200
@@ -94,11 +94,11 @@
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libdns.@SA@: ${OBJS}
+libdns.@SA@: touchfile ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libdns.la: ${OBJS}
+libdns.la: touchfile ${OBJS}
 	${LIBTOOL} --mode=link \
 		${CC} ${ALL_CFLAGS} -o libdns.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
@@ -124,7 +124,7 @@
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${GSSAPIOBJS} ${LIBS}
 
-timestamp: libdns.@A@
+timestamp: include libdns.@A@
 	touch timestamp
 
 installdirs:
@@ -172,3 +172,6 @@
 	include/dns/rdatastruct.h code.h
 subdirs: include/dns/enumtype.h include/dns/enumclass.h \
 	include/dns/rdatastruct.h code.h
+
+$(OBJS): include/dns/enumtype.h include/dns/enumclass.h \
+	include/dns/rdatastruct.h code.h
diff -urN bind-9.2.3.org/lib/isc/Makefile.in bind-9.2.3/lib/isc/Makefile.in
--- bind-9.2.3.org/lib/isc/Makefile.in	2003-10-24 23:31:34.000000000 +0200
+++ bind-9.2.3/lib/isc/Makefile.in	2003-10-24 23:31:50.000000000 +0200
@@ -88,11 +88,11 @@
 		-DLIBAGE=${LIBAGE} \
 		-c ${srcdir}/version.c
 
-libisc.@SA@: ${OBJS}
+libisc.@SA@: touchfile ${OBJS}
 	${AR} ${ARFLAGS} $@ ${OBJS}
 	${RANLIB} $@
 
-libisc.la: ${OBJS}
+libisc.la: touchfile ${OBJS}
 	${LIBTOOL} --mode=link \
 		${CC} ${ALL_CFLAGS} -o libisc.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
diff -urN bind-9.2.3.org/make/rules.in bind-9.2.3/make/rules.in
--- bind-9.2.3.org/make/rules.in	2003-10-24 23:31:35.000000000 +0200
+++ bind-9.2.3/make/rules.in	2003-10-24 23:31:50.000000000 +0200
@@ -47,7 +47,10 @@
 ### Makefile may define:
 ###	TARGETS
 
-all: subdirs ${TARGETS}
+all: touchfile ${TARGETS}
+
+touchfile: subdirs
+	touch touchfile
 
 ###
 ### Subdirectories
