diff -Nur bind-9.3.2.orig/configure.in bind-9.3.2.link/configure.in
--- bind-9.3.2.orig/configure.in	2006-08-05 01:45:48.000000000 +0200
+++ bind-9.3.2.link/configure.in	2006-08-05 01:45:55.000000000 +0200
@@ -1171,8 +1171,8 @@
 	*-linux*)
 		;;
 	*)
-		AC_CHECK_LIB(socket, socket)
-		AC_CHECK_LIB(nsl, inet_addr)
+		AC_SEARCH_LIBS(socket, socket)
+		AC_SEARCH_LIBS(inet_ntoa, nsl)
 		;;
 esac
 
diff -Nur bind-9.3.2.orig/lib/dns/Makefile.in bind-9.3.2.link/lib/dns/Makefile.in
--- bind-9.3.2.orig/lib/dns/Makefile.in	2004-12-09 05:07:15.000000000 +0100
+++ bind-9.3.2.link/lib/dns/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -39,7 +39,7 @@
 
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la -lcrypto
 
 # Alphabetically
 
diff -Nur bind-9.3.2.orig/lib/isccc/Makefile.in bind-9.3.2.link/lib/isccc/Makefile.in
--- bind-9.3.2.orig/lib/isccc/Makefile.in	2004-07-20 09:01:58.000000000 +0200
+++ bind-9.3.2.link/lib/isccc/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -36,7 +36,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCCDEPLIBS =	libisccc.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la
 
 SUBDIRS =	include
 
diff -Nur bind-9.3.2.orig/lib/isccfg/Makefile.in bind-9.3.2.link/lib/isccfg/Makefile.in
--- bind-9.3.2.orig/lib/isccfg/Makefile.in	2004-07-20 09:01:58.000000000 +0200
+++ bind-9.3.2.link/lib/isccfg/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -38,7 +38,7 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 ISCCFGDEPLIBS =	libisccfg.@A@
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ ../isc/libisc.la ../dns/libdns.la
 
 SUBDIRS =	include
 
diff -Nur bind-9.3.2.orig/lib/lwres/Makefile.in bind-9.3.2.link/lib/lwres/Makefile.in
--- bind-9.3.2.orig/lib/lwres/Makefile.in	2005-06-10 01:54:32.000000000 +0200
+++ bind-9.3.2.link/lib/lwres/Makefile.in	2006-08-05 01:45:55.000000000 +0200
@@ -46,7 +46,8 @@
 		lwinetaton.c lwinetpton.c lwinetntop.c print.c \
 		strtoul.c
 
-LIBS =		@LIBS@
+# -lpthread not needed here
+LIBS =
 
 SUBDIRS =	include man unix
 TARGETS =	timestamp
