--- src/lib/isc/eventlib.c	Sat Mar 21 00:26:24 1998
+++ src.orig/lib/isc/eventlib.c	Sat Dec  5 17:43:14 1998
@@ -42,7 +42,11 @@
 #include "port_after.h"
 
 /* Forward. */
-
+#if ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 1))
+#include <sys/select.h> /* Another piece of shit from glibc guys ...;(( */
+#undef NEED_PSELECT 
+#endif /* glibc-2.1 */
+ 
 #ifdef NEED_PSELECT
 static int		pselect(int, void *, void *, void *, struct timespec*);
 #endif
