--- bind-9.18.9/lib/dns/zone.c~	2022-11-07 23:17:42.000000000 +0100
+++ bind-9.18.9/lib/dns/zone.c	2022-12-20 11:06:25.886493925 +0100
@@ -5801,7 +5801,7 @@ dns_zone_detach(dns_zone_t **zonep) {
 		 * Stop things being restarted after we cancel them below.
 		 */
 		DNS_ZONE_SETFLAG(zone, DNS_ZONEFLG_EXITING);
-		dns_zone_log(zone, ISC_LOG_INFO, "final reference detached");
+		dns_zone_log(zone, ISC_LOG_DEBUG(1), "final reference detached");
 		if (zone->task != NULL) {
 			/*
 			 * This zone has a task; it can clean
