--- bind-9.20.1/lib/isc/mem.c~	2024-08-13 17:25:55.000000000 +0200
+++ bind-9.20.1/lib/isc/mem.c	2024-08-21 15:21:51.356574025 +0200
@@ -48,6 +48,7 @@
 #include <malloc_np.h>
 #define JEMALLOC_API_SUPPORTED 1
 #elif defined(HAVE_JEMALLOC)
+#define JEMALLOC_MANGLE 1
 #include <jemalloc/jemalloc.h>
 #define JEMALLOC_API_SUPPORTED 1
 #else
--- bind-9.18.28/lib/isc/include/isc/mem.h~	2024-07-08 17:20:32.000000000 +0200
+++ bind-9.18.28/lib/isc/include/isc/mem.h	2024-07-23 22:40:27.816664863 +0200
@@ -189,7 +189,7 @@ extern unsigned int isc_mem_defaultflags
  * see https://github.com/jemalloc/jemalloc/issues/2566
  * for more information.
  */
-#if HAVE_JEMALLOC
+#if HAVE_JEMALLOC && 0
 
 /*
  * cmocka.h has confliction definitions with the jemalloc header but we only
